### ISSUE TITLE
Rocket Assembling Machine - Blank Screen

### ISSUE THREAD

**Issue by LokeYourLord** [zmaster587/AdvancedRocketry#1107]
Minecraft Version:
1.12.2
Forge Version:
14.23.1.2555
Advanced Rocketry Version:
1.2.6.-52
LibVulpes Version:
0.2.7.-24
**Is this the latest curseforge build?**
Yes, it is.
Crash Report:
None
Explain issue in detail:
Basically, the issue appears when I either want to scan a Rocket or Build a Rocket. The second I click either of those Buttons, my Screen goes blank, however, I noticed that once the Rocket Assembling Machine block gets destroyed, that the bug disappears (that can happen by clicking ESC and breaking the block in-front of you even if you can't see, or anything else breaking it, players, explosions, whatever).

I will post a short video in order to properly explain the situation (soon).

EDIT: I did a little test with a much smaller Launchpad (the smallest) and tried to scan, apparently, it ONLY happens when that Scan/Building Animation is going on, once it finishes, it returns back to the normal interface. It also works clicking ESC and looking away from the Launchpad, as long as you don't see the Scan/Building Animation, you will have a clear view of everything.

I can provide a list of mods installed on my client, here you go:
```
	| State | ID                      | Version                  | Source                                          | Signature                                |
	|:----- |:----------------------- |:------------------------ |:----------------------------------------------- |:---------------------------------------- |
	| UCH   | minecraft               | 1.12.2                   | minecraft.jar                                   | None                                     |
	| UCH   | mcp                     | 9.42                     | minecraft.jar                                   | None                                     |
	| UCH   | FML                     | 8.0.99.99                | modpack.jar                                     | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCH   | forge                   | 14.23.1.2555             | modpack.jar                                     | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCH   | advancedrocketrycore    | 1                        | minecraft.jar                                   | None                                     |
	| UCH   | smoothfontcore          | 1.12.2-1.0               | minecraft.jar                                   | None                                     |
	| UCH   | gottagofast             | 1.0                      | gottagofast-1.0.jar                             | None                                     |
	| UCH   | smoothfont              | 1.12.2-1.2               | SmoothFont-1.12.2-1.2.jar                       | None                                     |
	| UCH   | securitycraft           | v1.8.3                   | [1.12.x] SecurityCraft v1.8.3.jar               | None                                     |
	| UCH   | redstoneflux            | 2.0.1                    | RedstoneFlux-1.12-2.0.1.2-universal.jar         | d4f1503fbacd9b9fb767720420c5395104239ec9 |
	| UCH   | cofhcore                | 4.3.7                    | CoFHCore-1.12.2-4.3.7.25-universal.jar          | None                                     |
	| UCH   | libvulpes               | 0.2.7.-24                | LibVulpes-1.12.2-0.2.7-24-universal.jar         | None                                     |
	| UCH   | advancedrocketry        | 1.2.6.-52                | AdvancedRocketry-1.12.2-1.2.6-52.jar            | None                                     |
	| UCH   | appliedenergistics2     | rv5-stable-3             | appliedenergistics2-rv5-stable-3.jar            | None                                     |
	| UCH   | bdlib                   | 1.14.3.12                | bdlib-1.14.3.12-mc1.12.2.jar                    | None                                     |
	| UCH   | ae2stuff                | 0.7.0.4                  | ae2stuff-0.7.0.4-mc1.12.2.jar                   | None                                     |
	| UCH   | jei                     | 4.8.5.138                | jei_1.12.2-4.8.5.138.jar                        | None                                     |
	| UCH   | appleskin               | 1.0.9                    | AppleSkin-mc1.12-1.0.9.jar                      | None                                     |
	| UCH   | attributefix            | 1.0.1                    | AttributeFix-1.12.2-1.0.1.jar                   | d476d1b22b218a10d845928d1665d45fce301b27 |
	| UCH   | bedrockreplacer         | 7.0.2                    | BedrockReplacer-Forge-7.0.2.jar                 | None                                     |
	| UCH   | betterplacement         | 1.0.0-1                  | BetterPlacement-1.0.0-1.jar                     | None                                     |
	| UCH   | camoucreepers           | 1.5                      | camoucreepers-1.5.jar                           | None                                     |
	| UCH   | clienttweaks            | 3.1.2                    | ClientTweaks_1.12-3.1.2.jar                     | None                                     |
	| UCH   | codechickenlib          | 3.1.5.331                | CodeChickenLib-1.12.2-3.1.5.331-universal.jar   | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| UCH   | cofhworld               | 1.0.2                    | CoFHWorld-1.12.2-1.0.2.10-universal.jar         | None                                     |
	| UCH   | cyberware               | 0.2.10.5                 | cyberware-1.12.2-0.2.10.5.jar                   | 67a0e286dc0d4b502f3c92ac20b953517b52d0a9 |
	| UCH   | dimensionloader         | 1.12.2-1.0               | dimensionloader-1.12.2-1.0.jar                  | None                                     |
	| UCH   | discordsuite            | 1.1.4                    | DiscordSuite-1.1.4.jar                          | None                                     |
	| UCH   | ecomod                  | 1.12.2-1.4.1.0           | EcologyMod-1.12.2-1.4.1.0.jar                   | None                                     |
	| UCH   | enderstorage            | 2.4.2.126                | EnderStorage-1.12.2-2.4.2.126-universal.jar     | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| UCH   | waila                   | 1.8.23                   | Hwyla-1.8.23-B38_1.12.jar                       | None                                     |
	| UCH   | extracells              | 2.5.10                   | ExtraCells-1.12.2-2.5.10a47.jar                 | None                                     |
	| UCH   | felling                 | 1.3.2                    | felling-1.3.2-1.12.2.jar                        | None                                     |
	| UCH   | forgelin                | 1.6.0                    | Forgelin-1.6.0.jar                              | None                                     |
	| UCH   | forgivingvoid           | 1.0.5                    | ForgivingVoid_1.12-1.0.5.jar                    | None                                     |
	| UCH   | grue                    | 1.3.5                    | Grue-1.12-1.3.5.jar                             | None                                     |
	| UCH   | hammercore              | 1.9.5.7                  | HammerCore-1.12.2-1.9.5.7.jar                   | None                                     |
	| UCH   | hcisland                | 0.2.2                    | hcisland-0.2.2.jar                              | None                                     |
	| UCH   | ichunutil               | 7.1.4                    | iChunUtil-1.12.2-7.1.4.jar                      | None                                     |
	| UCH   | igwmod                  | 1.4.3-14                 | IGW-Mod-1.12.2-1.4.3-14-universal.jar           | None                                     |
	| UCH   | mekanism                | 1.12.2-9.4.3.330         | Mekanism-1.12.2-9.4.3.330.jar                   | None                                     |
	| UCH   | tesla                   | 1.0.63                   | Tesla-1.12.2-1.0.63.jar                         | d476d1b22b218a10d845928d1665d45fce301b27 |
	| UCH   | teslacorelib            | 1.0.12                   | tesla-core-lib-1.12-1.0.12.jar                  | None                                     |
	| UCH   | industrialforegoing     | 1.12.2-1.12.2            | industrialforegoing-1.12.2-1.7.0-114.jar        | None                                     |
	| UCH   | inventorytweaks         | 1.63+release.109.220f184 | InventoryTweaks-1.63.jar                        | 55d2cd4f5f0961410bf7b91ef6c6bf00a766dcbe |
	| UCH   | journeymap              | 1.12.2-5.5.2             | journeymap-1.12.2-5.5.2.jar                     | None                                     |
	| UCH   | jeresources             | 0.8.5.27                 | JustEnoughResources-1.12.2-0.8.5.27.jar         | None                                     |
	| UCH   | longfallboots           | 1.2.1a                   | longfallboots-1.2.1a.jar                        | None                                     |
	| UCH   | lostcities              | 1.0.2                    | lostcities-1.12-1.0.2.jar                       | None                                     |
	| UCH   | mekanismgenerators      | 9.4.3                    | MekanismGenerators-1.12.2-9.4.3.330.jar         | None                                     |
	| UCH   | mekanismtools           | 9.4.3                    | MekanismTools-1.12.2-9.4.3.330.jar              | None                                     |
	| UCH   | minecolonies            | 1.12.2-0.8.7210          | minecolonies-universal-1.12.2-0.8.7210.jar      | None                                     |
	| UCH   | minemenu                | 1.6.3                    | MineMenu-1.12.2-1.6.3-universal.jar             | None                                     |
	| UCH   | mousetweaks             | 2.8                      | MouseTweaks-2.8-mc1.12.1.jar                    | None                                     |
	| UCH   | netherportalfix         | 5.3.13                   | NetherPortalFix_1.12.1-5.3.13.jar               | None                                     |
	| UCH   | nuclearcraft            | 2.9b                     | NuclearCraft-2.9b--1.12.2.jar                   | None                                     |
	| UCH   | thermalfoundation       | 2.3.7                    | ThermalFoundation-1.12.2-2.3.7.18-universal.jar | None                                     |
	| UCH   | thermalexpansion        | 5.3.7                    | ThermalExpansion-1.12.2-5.3.7.31-universal.jar  | None                                     |
	| UCH   | omlib                   | 3.0.0-133                | omlib-1.12.2-3.0.0-133.jar                      | None                                     |
	| UCH   | openmodularturrets      | 3.0.0-230                | openmodularturrets-1.12.2-3.0.0-230.jar         | None                                     |
	| UCH   | portalgun               | 7.0.0                    | PortalGun-1.12.2-7.0.1.jar                      | None                                     |
	| UCH   | redstonearsenal         | 2.3.7                    | RedstoneArsenal-1.12.2-2.3.7.18-universal.jar   | None                                     |
	| UCH   | redstonepaste           | 1.7.5                    | redstonepaste-mc1.12-1.7.5.jar                  | None                                     |
	| UCH   | refinedrelocation       | 4.2.16                   | RefinedRelocation_1.12.2-4.2.16.jar             | None                                     |
	| UCH   | servertickmonitor       | 3.0.0                    | ServerTickMonitor-1.12-3.0.0.jar                | None                                     |
	| UCH   | sgcraft                 | 1.13.2                   | SGCraft-1.14.2-mc1.12.2.jar                     | None                                     |
	| UCH   | soulshardstow           | 1.12-2.7.4-53            | SoulShards-TOW-1.12-2.7.4-53.jar                | None                                     |
	| UCH   | sync                    | 7.0.0                    | Sync-1.12.2-7.0.1.jar                           | None                                     |
	| UCH   | teslafied               | 1.1.1                    | teslafied-1.12-1.1.1.jar                        | None                                     |
	| UCH   | beneath                 | 1.3.0                    | The Beneath-1.12.2-1.3.0.jar                    | 220f10d3a93b3ff5fbaa7434cc629d863d6751b9 |
	| UCH   | thermalcultivation      | 0.1.2                    | ThermalCultivation-1.12.2-0.1.2.8-universal.jar | None                                     |
	| UCH   | thermaldynamics         | 2.3.7                    | ThermalDynamics-1.12.2-2.3.7.14-universal.jar   | None                                     |
	| UCH   | mysttmtgravitymod       | 2.8.0                    | UpAndDownAndAllAround-1.12.2-2.8.0.jar          | None                                     |
	| UCH   | wailaharvestability     | 1.1.12                   | WailaHarvestability-mc1.12-1.1.12.jar           | None                                     |
	| UCH   | wanionlib               | 1.12.2-1.4               | WanionLib-1.12.2-1.4.jar                        | None                                     |
	| UCH   | wawla                   | 2.5.246                  | Wawla-1.12.2-2.5.246.jar                        | d476d1b22b218a10d845928d1665d45fce301b27 |
	| UCH   | wearablebackpacks       | 3.1.0                    | WearableBackpacks-1.12.2-3.1.0.jar              | None                                     |
	| UCH   | xray                    | 1.4.0                    | xray-1.12.2-1.4.0.jar                           | None                                     |
	| UCH   | teslacorelib_registries | 1.0.12                   | tesla-core-lib-1.12-1.0.12.jar                  | None                                     |
	| UCH   | unidict                 | 1.12.2-2.1               | UniDict-1.12.2-2.1.jar                          | None                                     |
```

**Comment by zmaster587** [comment]
I don't recognise any mods on the list that may do this.  I know optifine and betterfps both have a habit of messing with the scanner.  If i were to take a guess, maybe it's an interaction with xray, clienttweaks, or smoothfont.  Can you confirm which, if any it is, if it's none of them, can you try to isolate the minimum set of mods required to replicate the issue?

**Comment by LokeYourLord** [comment]
Alright, will do, has been a while since I have been doing these things so I forgot that I should isolate the problem first with any client side graphical mods :P Thanks, I will remove them one by one and see when it stops occurring.

**Comment by LokeYourLord** [comment]
Alright, even though you mentioned Optifine (and I knew it _could_ be that), and I excluded it first, I half way through the mods list tried to remove it, instantly fixed it, so it's Optifine. I also tried without ALL other mods besides Advanced Rocketry, LibVulpes and Optifine, and it causes it. And just to be 100% sure, I added all mods BESIDES Optifine back, but the issue doesn't occur then.

Short version: It's Optifine, nothing else causes this.

**Comment by nmarshall23** [comment]
When Betterfps options for fast fog or fast beacon beam are enabled, I get a Blank Screen when looking at a Rocket Assembling Machine that is scanning. 

Changing those options back to fancy fixed the problem.

**Comment by AnthyG** [comment]
@nmarshall23 the solution proposed by you works like a charm!

**Comment by dmodoomsirius** [comment]
Is this still a issue? Please comment if it is and i will reopen if if you are unable? We are closing old issues to see what is still relevant and needs to be fixed.
